### DISCUSSION TITLE
Weekly Blog updates

### DISCUSSION THREAD

**Discussion by Qiriba** [Qiriba/Rezepte-for-Dummies#4] [Announcements]
Weekly Blog Updates

**Comment by Qiriba** [comment]
![grafik](https://github.com/Qiriba/Rezepte-for-Dummies/assets/136338149/9723bb48-3736-4f75-8536-2aac401c1af4)

Woche 2

**Comment by Qiriba** [comment]
Nachtrag für Woche 1:


In der ersten Woche haben wir folgende Aufgaben erledigt:

- Projekt vision festgelegt
- Tech Stack soweit wie geht festgelegt
- Projektmanagment-Platform ausgesucht
- Quellcoderepository festgelegt
- Blog Kanal eröffnet
- Team Informationen bekannt gegeben

**Comment by 999Marlon** [comment]
Woche 3:

- UI-Mockups sind erstellt
- Softwarespezifikationsanalyse ist erstellt und ist im Github unter dem Ordner Aufgaben hochgeladen
- UML Anwendungsfall ist erstellt
-
<img width="417" alt="Woche 3" src="https://github.com/Qiriba/Rezepte-for-Dummies/assets/137040652/6b2cef71-afc3-4f7c-bb8a-38fbae6f8814">

**Reply by Bjoern04** [reply]
SRS sieht gut aus, jedoch kann man das UI-Mockup nicht anschauen. Man könnte noch zum Geltungsbereich etwas mehr Informationen hinzufügen.

**Reply by Qiriba** [reply]
Um das UI-MockUp anschauen zu können einfach Access requesten, ich gebe diesen dann

**Comment by cool2722** [comment]
1) Du kannst ein View-Only Link erstellen in Figma (noch nicht reviewed).

2) Unter Funktionale Anforderungen,

- Bitte "Materialien" genauer definieren - Zutaten, Kochutensilien?
- Die letzte zwei punkten gehören nicht zu Funktionale-Anforderungen. (sind schon in Nicht-Funktionale Anforderungen bzw. Technische Einschränkungen.

**Comment by 999Marlon** [comment]
<img width="351" alt="Woche 4" src="https://github.com/Qiriba/Rezepte-for-Dummies/assets/137040652/2e9c0a3e-7f7e-47f7-9298-f505a392a7d0">


- Aktuelles SRS Dokument in PDF Form
[SRS PDF.pdf](https://github.com/Qiriba/Rezepte-for-Dummies/files/13215332/SRS.PDF.pdf)

- UML - Diagramme sind erstellt und im SRS Word Dokument hinterlegt
- Wir benutzen nun Word.io anstelle von Lucidcharts

**Reply by Bjoern04** [reply]
Aktivitätsdiagramme starten mit einem ausgefüllten Punkt und nicht mittendrin. Ende eines Aktivitätsdiagramms wird mit zwei Kreisen symbolisiert. Dies sollte noch verbessert werden.

**Reply by cool2722** [reply]
Da fehlt noch ein Aktivitätsdiagramm (der Zweite ist nur kopiert).
Beim zweiten Akitvitätsdiagramm, nach dem Schritt "Spiele die Audiodatei für die aktuelle Seite ab" , geht der Pfeil in zwei Richtungen ohne eine 'Oder Bedingung'.

**Reply by 999Marlon** [reply]
@cool2722 Den ersten Punkt von dir können wir nicht ganz nachvollziehen. Den Zweiten Punkt haben wir behoben

**Reply by hmi95** [reply]
Gibt es einen Unterschied zwischen den Aktivitätsdiagrammen unter 2.3 und 2.5?
Seitenzahlen im Dokument wären nützlich, dann könnte ich diese hier auch als Referenz angeben.
Wo finde ich das aktualisierte SRS Dokument? Ist es in Git versioniert?

**Comment by 999Marlon** [comment]
<img width="581" alt="Woche 5" src="https://github.com/Qiriba/Rezepte-for-Dummies/assets/137040652/c84be92b-6722-4040-9377-b106c98dc41a">

- Sprint 3 ist gestartet
- Vorläufiges UML - Klassendiagramm ist erstellt und im SRS Dokument hinterlegt worden
-> Falls Fragen zu den Klassen gibt, gerne melden!
- Konstruktoren sind erstellt worden

<img width="459" alt="CurrentStep" src="https://github.com/Qiriba/Rezepte-for-Dummies/assets/137040652/089702c0-5a70-4c5f-bfad-81891c4132e8">

**Reply by Bjoern04** [reply]
Hat es einen Grund, dass ihr bei "Ingredient" für "amount" den Typ String genommen habt? Würde dort ein Double nicht besser passen? Sonst sieht alles gut und nachvollziehbar aus.

**Reply by Qiriba** [reply]
Wir haben uns absichtlich dafür entschieden String zu nehmen da Mengenangaben bei Rezepten häufig nicht nur in ml angegeben sind. So wird zum Beispiel häufig Mengenangaben in Esslöffeln oder ähnlichen angegeben. Außerdem werden in Ländern wie Amerika ausschließlich die einheit "CUPS" verwendet. Wenn wir also die Anzahl als String abspeichern können wir besser und exacter angeben was die Richtige einheit ist. Danke für das Feedback 👍

**Reply by cool2722** [reply]
Falls die Mengenangabe statisch ist, dann ist es ok die Menge als String zu speichern aber falls mit dem Zahlenwert gerechnet werden muss, z.B. wenn man beim Rezept die doppelte Menge braucht wenn man 2 Portionen eines Gerichts zubereiten möchte, dann ist es vom Vorteil die Menge als Zahl zu speichern und zusätzlich noch ein weiteres Attribut für die Einheit zu speichern.

Ansonsten sieht alles gut aus!

**Reply by Qiriba** [reply]
Das ist ein sehr gutes argument. Wir überlegen uns nochmal ob es eine bessere Implementierung gibt. Vielen Dank für das Feedback

**Reply by FabianDev001** [reply]
Je nachdem wie euch der Aufwand wert ist, würde sich vielleicht auch eine zusätzliche Klasse für Ingredienz lohnen, im welchem ihr den Wert (als Zahl und zum Anzeigen in Textform) und (Funktionen/) Umrechnungen für die jeweiligen Typen zur Verfügung stellen könntet. 

Falls ihr später vorhaben solltet, Rezepte nach einem Filter anzuzeigen, wäre es auch eine Überlegung "difficulty" als (Ganz-)Zahl (bspw. 1-3) abzuspeichern. [(als Beispiel)](https://www.rewe.de/rezepte/).

Sowie fällt mir auf das ihr kein Passwort in euerem Klassendiagramm aufgeschrieben habt... Speichert ihr dieses dann in einer Datenbank? (frage aus Neugier meinerseits).

Letztendlich, props an euer Klassendiagramm, es ist euch wirklich gut gelungen.

**Reply by 999Marlon** [reply]
@FabianDev001 Das ist sehr gute Rückmeldung, danke Kai :)

**Comment by HabubMarcel** [comment]
# Blogeintrag Woche 6
* Hello World Demo wurde erstellt
* Android Studio ist bei allen funktionsfähig eingerichtet und die Demo läuft bei jedem

<img width="227" alt="image" src="https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/03b5452d-45d7-450b-a6cc-b6d9942074ac">

### Das ganze wird aus folgendem Quellcode generiert:

![Snap](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/0d10ad6a-e810-468e-b5ec-923af29d849d)

* Den Code findet ihr im folgenden Repository: Der Code für die Android App wird auf dieser Seite geupdatet https://github.com/Qiriba/Recipesfordummiescode/tree/master
* Das Github und der Code wurden von Tobias eingerichtet

### Das ist das Kanban Board für diese Woche:

![SPrint](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/fd904750-92f9-4d43-a5ae-4790eff227f8)

**Reply by JanWilfert** [reply]
Ich habe ein paar Anmerkungen/Fragen:
- Warum sind eure tasks keiner Person zugewiesen? Heißt das, dass jeder alle Aufgaben gemacht hat? Wie habt ihr vor am Ende auszuwerten wie lange jeder in welchem Bereich gearbeitet hat? Es wäre vielleicht hilfreich jeder Tasks eine Person und zusätzlich die benötigte Zeit für jede Task aufzuschreiben.
- Warum ist euer Quellcode in einem anderem github repo?

Insgesamt sieht das dann alles auch ganz gut aus. Eure Demo zeigt, dass ihr fähig seid eine Hello World Android App zu entwickeln.

**Reply by okramDE** [reply]
Hallo,
Gut das die Android Umgebung schonmal funktioniert. Wir wollten nur fragen wie ihr eure Daten speichert wenn zu User-Daten kommt da diese ja nicht gehardcoded werden können.
Wir sind auf die kommenden Wochen gespannt!

**Comment by HabubMarcel** [comment]
# Woche 7
### User Story wurde implementiert: "Der Benutzer soll zwischen verschiedenen Rezepten auswählen können"


https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/970c6ef1-53de-4a22-a10f-b33c9e099709


### XML App Layout

#### Grundseite:

<img width="501" alt="xml2" src="https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/a4622fba-3d08-447e-8b90-f4c98f88efaa">

#### Items die auf der Grundseite liegen:

<img width="508" alt="xml1" src="https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/105eef95-da88-4e3c-8736-8697e639bd64">

**Comment by Bjoern04** [comment]
Sieht soweit gut aus. Eine Frage habe ich noch. Aus den dargestellten Materialien geht leider nicht hervor, ob der Nutzer die einzelnen Rezepte wirklich auswählen kann und Informationen dazu bekommt, oder ob dieser "nur" durch die verschiedenen Gerichte scrollen kann.

**Reply by Qiriba** [reply]
Die User story war nur das der Benutzer zwischen den verschiedenen Rezepten auswählen kann. Wir haben weitere user storys die eine detaillierte Rezeptseite enthalten. Dementsprechend sind die einzelnen Rezepte noch nicht anklickbar

**Comment by amjadalnakshbandi** [comment]
Hallo, das sieht sehr gut und lecker aus.
Habt ihr noch vor, einen Filter für die Liste zum Implementieren?

Wir sind auf die kommenden Wochen gespannt!

**Reply by Qiriba** [reply]
* Um genaue Rezepte auswählen zu können werden wir wohl eine Suchfunktion implementieren müssen. Diese wird dann auch eine Filterfunktion haben. 
* Die aktuelle Implementation ist eher wie ein Feed, ähnlich zu Twitter. Aktuell haben wir noch keine Implementation um den Feed von einem Server zu füllen, d.h. sobald wir diese Einbindung haben sollte es auch kein großer Aufwand sein einen Filter für den Feed einzubauen.

### Vielen Dank aber für die Idee, haben wir mit aufgenommen

**Comment by HabubMarcel** [comment]
# Blogeintrag Woche 8
## Qualitätsbaum erstellt
<img width="501" alt="asr" src="https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/9d69e7dd-31fd-48e5-9ed7-66de5a90f24f">

## Zusammenfassung von Architekturentscheidungen und Designmustern:

[ARS_PDF.pdf](https://github.com/Qiriba/Rezepte-for-Dummies/files/13491257/ARS_PDF.pdf)

## Firebase
* Firebase als Backend-Managementsystem um Daten in Echtzeit abrufen zu können
* Zusätzlich wird Performance und Sicherheit gewährleistet

## Picasso
[Link zu Picasso](https://square.github.io/picasso/)

* Wir benutzen Picasso um Bilder aus dem Internet zu laden
* Picasso unterstützt caching und asynchrones laden von Bildern
![picasso](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/6400905c-0016-417a-8919-4aedfed682fe)

**Reply by Bjoern04** [reply]
Guter Blogeintrag! Er ist diese Woche sehr ausführlich. Bei den Szenarien für die Qualitätsattribute könntet ihr spezifisch, messbare Ziele dazuschreiben. Also genaue Ziele, welche im Nachhinein durch einfaches Abmessen überprüft werden können. Beispielweise beim Leistungsattribut könntet ihr hinzufügen, in wie vielen Sekunden die App gestartet werden soll. Oder bei der Zuverlässigkeit, dass eure App in 99% der Fälle eine konsistent Datenverarbeitung ohne Ausfälle vorweisen kann.

**Comment by FabianDev001** [comment]
Moin, euer Blockpost für diese Woche ist wieder wirklich ausführlich worden, chapeau!

Zu eurem Qualitätsbaum kommen mir keine weiteren Fragen auf...

Was mich interessieren würde, wäre wieso ihr euch genau für Picasso entscheiden habt?, und keine Alternative...?
Bzw. war das "caching und asynchrones laden von Bildern" euer einziger Hauptgrund?

Ansonsten finde ich eure Idee mit Firebase super, und bin selbst gespannt wie ihr dies in euer Projekt einbaut.

**Reply by Qiriba** [reply]
#### Neben Picasso gibt es viele von diesen Image Loading Libraries wie z.B. Glide, Fresco oder Coil. Diese nehmen sich im allgemeinen nicht sehr viel, die wichtigsten Features die wir identifiziert haben waren: 

* Image Caching
* Asynchrones Laden
* Integrier und Benutzaufwand
* Bild Transformations unterstützung
* Performance

Picasso erfüllt all diese Anforderungen, weshalb wir uns für Picasso entschieden haben, Glide wäre aber genau so gegangen. Danke für den Kommentar, wir sind auch sehr gespannt auf Firebase, in der Theorie sollte Firebase all unsere Backend Anforderungen vereinfachen, aber das werden wir wohl noch sehen

**Comment by HabubMarcel** [comment]
# Blogeintrag Woche 9
## Arc42
Unsere Arc42 PDF: [arc42.pdf](https://github.com/Qiriba/Rezepte-for-Dummies/files/13579799/arc42.pdf)

Ich hoffe unser Arc42 Dokument entspricht den Anforderungen, falls nicht gerne in die Kommentare schreiben, dann verbessern wir das in Woche 10.

**Reply by Bjoern04** [reply]
Sehr knapp. Es sollten mehrere Architekturentscheidungen aufgeführt werden. Zu den Qualitätszielen könnten noch die passenden Qualitätsattribute dazu geordnet werden.

**Reply by JulianHeckel** [reply]
Euer Dokument ist sehr kurzgehalten. Mich würde vor allem noch eure Aufgabenstellung sowie eure Kontextabgrenzung interessieren.
Wie organisiert ihr euch im Team? 
Habt ihr außer der Architektur schon irgendwelche Grundlagen festgelegt, zum Beispiel welche Programmiersprache ihr verwendet? 
Ihr habt zwar eure Qualitätsziele angegeben, aber mich würde noch interessieren wie ihr diese Ziele erreichen, bzw. garantieren wollt.
Ich finde es sehr gut, dass ihr übersichtliche Diagramme eingefügt habt, aber ich fände eine kurze Erklärung was man in den Diagrammen sehen kann, bzw. zu welchem Zweck ihr sie dort aufgeführt habt sehr hilfreich.
Ich finde es gut, dass eure Qualitätsanforderungen sehr ausführlich und verständlich formuliert sind.
Alles in allem ein sehr knapp ausgefülltes Arc42 Dokument, wo ich mir noch einige Ergänzungen wünschen würde.

**Comment by amjadalnakshbandi** [comment]
Hallo, 
Es sieht sehr gut aus. 
Wir haben eine paar Fragen: 
-  Was ist die Komponente "Storage" genau? 
-  Habt ihr ein Design-Pattern für eure Architekturentscheidungen? 

Wir sind auf die nächste Woche gespannt.

**Comment by 999Marlon** [comment]
# Blogeintrag Woche 10
## Willkommen zu unserem letztem Blogeintrag zu diesem Semester

[Hier findet ihr unseren Link zur SRS](https://github.com/Qiriba/Rezepte-for-Dummies/blob/main/Aufgaben/Software%20Anforderungen%20Spezifikation.docx)

[Hier findet ihr unseren Link zum ARC42](https://github.com/Qiriba/Rezepte-for-Dummies/blob/main/Aufgaben/ARC42_PDF.pdf)

Wir wünschen euch Frohe Weihnachten und einen guten Rutsch! 🎅🏿🎄

**Comment by HabubMarcel** [comment]
# Blogeintrag Woche 2 Semester 4
## Willkommen zu unserem ersten Blogeintrag in diesem Semester
### RMMM-Tabelle
[RMMM-Tabelle](https://docs.google.com/spreadsheets/d/1EHs1eGxeq9m9_02dJjc-QpRDIaq5jex5JFTOdLCUm8g/edit?pli=1#gid=0)</br>
Wir haben unserer RMMM in Excel erstellt.
Unser größtes Risiko ist der Mitarbeiterausfall, da wir wenig Arbeiter haben und dadurch einzelne ausfälle Problematisch werden können --> Daher Isolieren wir unsere Mitarbeiter und achten darauf, dass sie immer Impftechnisch up to date sind. Falls es zu ausfällen kommt, müssen die Mitarbeiter Überstunden machen.

**Reply by amjadalnakshbandi** [reply]
Hallo zusammen,

ich finde den Blogeintrag insgesamt zufriedenstellend. Die Auflistung und Kategorisierung der Risiken nach Schadensausmaß und Eintrittswahrscheinlichkeit halte ich für gut durchdacht. Beim ersten Risiko habt ihr jedoch keine Minimierungsstrategie genannt. Als Vorschlag würde ich empfehlen, ein Datenbank-Backup zu erstellen und dieses als Notfallplan zu nutzen.

Eine kleine Korrektur: Bei der Erwähnung von "Firebase" scheint ein Tippfehler vorzuliegen.

Abgesehen davon habe ich keine weiteren Fragen oder Anmerkungen und freue mich auf den nächsten Beitrag.

**Comment by HabubMarcel** [comment]
# Blogeintrag Woche 3 Semester 4
##  Code Refactoring
Willkommen zu unserem Blogeintrag.
Diese Woche haben wir uns mit dem Code Refactoring befasst.
Wir haben bis dato sehr darauf geachtet nach den Allgemeinen Clean Code Prinzipien zu programmieren,
dennoch gab es wenige Verbesserungs Möglichkeiten, die ihr euch im Bild das sich unter diesem Text befindet anschauen könnt.

**Verbesserte Lesbarkeit:** Durch die Anpassung der Namenskonventionen und die aussagekräftige Benennung der Variablen wird der Code leichter zu lesen und zu verstehen. 

**Klare Bedeutung der Variablen:** Die Umbenennung der Variablen vermittelt eine klare Bedeutung ihres Zwecks, was die Absicht des Codes transparenter macht. 

**Einheitliche Struktur:** Die konsistente Benennung der Variablen stellt sicher, dass der Code eine einheitliche Struktur aufweist, was wiederum die Wartbarkeit verbessert. 

**Entfernung von Redundanzen:** Durch das Entfernen überflüssiger Leerzeichen wird der Code sauberer und konsistenter gestaltet, was auch die Lesbarkeit fördert. 

**Bessere Wartbarkeit:** Die durchgeführten Refaktorisierungen zielen darauf ab, den Code wartbarer zu machen, indem sie eine klare Struktur schaffen und sicherstellen, dass zukünftige Änderungen leichter durchgeführt werden können. 
![Code Refactoring](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/71b9f9ff-27fa-499c-a845-4104c46c7807)

**Reply by Bjoern04** [reply]
Das sieht sehr gut aus. Eure Änderungen ergeben Sinn und sind nachvolllziehbar. Die Variablennamen sind nun deutlich aussagekräftiger und einheitlich in englisch verfasst.

**Reply by cool2722** [reply]
Wir haben die gleiche Meinung wie Bjoern aber ich wundere mich warum habt ihr 'tv' als ein Präfix für die Variable Name? Ist es nicht Redundant oder gehört dies zur Vereinheitlichung?

**Reply by Qiriba** [reply]
tv steht für textView, diese Variablen sind also mit dem textview direkt in Verbindung, ist natürlich für einen Laien nicht erkennbar

**Comment by 999Marlon** [comment]
# Blogeintrag Woche 4 Semester 4
## Testplan
- Wir verwenden das JUnit-Test-Framework für unsere Unit Tests. Derzeit haben wir keine Pläne für instrumentierte Tests, aber wir planen, später UI-Tests durchzuführen. Aktuell gibt es dafür noch nicht genügend testbare UI-Elemente [Was sind Instrumentierten Tests ](https://developer.android.com/training/testing/instrumented-tests?hl=de).
- Wir wollen eine Unit Test Abdeckung von ca. 60% erreichen. 
- Als automatisierte Testwerkzeuge benutzen wir GitHub Actions.
- Zur Verfolgung der Testergebnisse können wir in Android Studio den Testbericht anzeigen, der nach dem Ausführen der Tests generiert wird. Dieser Bericht zeigt an, welche Tests bestanden und welche fehlgeschlagen sind.
---
### Unit Test
Bei der Erstellung des Projektes sind Unit Tests standartmäßig bereits erstellt worden.
Hier ist ein Implemetierter Test in unserem Code

![UnitTest](https://github.com/Qiriba/Rezepte-for-Dummies/assets/137040652/7b5ba556-a047-4787-9fbc-788a7d667cf2)

**Reply by amjadalnakshbandi** [reply]
Die angestrebte Testabdeckung erscheint sowohl sinnvoll als auch realistisch. Es ist auch super, dass ihr GitHub Actions zur Nachverfolgung der Tests einsetzt. Das dürfte die Qualitätssicherung effektiv unterstützen.
Plant ihr, die Testabdeckung in Zukunft weiter zu erhöhen?
Ich freue mich auf den nächsten Beitrag.

**Reply by Bjoern04** [reply]
Der Testplan den ihr entwickelt habt sieht soweit gut aus. 
Es ist auch gut, dass ihr euch messbare Ziele gesetzt habt, die ihr nachher einfach überprüfen könnt.
Die Verfolgung der Testergebnisse über Adroid Studio ist sinnvoll.

**Comment by HabubMarcel** [comment]
# Blogeintrag Woche 5 Semester 4
## Hinzufügen von CI/CD Pipeline und CLickable Objects
### CI/CD Pipelines 
Um unsere Unittests zu verbessern haben wir nun 2 Pipelines. Die eine baut das Projekt und schaut das dort alles sauber abläuft. Auf der anderen Pipeline werden die Unittests ausgeführt. Wir haben uns außerdem dafür entschieden, weil CI/CD die Zusammenarbeit innerhalb des Entwicklungsteams fördert, da Änderungen kontinuierlich integriert und bereitgestellt werden. Dadurch wird die Kommunikation verbessert und das Risiko von Konflikten minimiert. Insgesamt bieten CI/CD Pipelines eine effiziente und zuverlässige Möglichkeit, Software zu entwickeln, zu testen und bereitzustellen, was zu einer verbesserten Produktqualität und einer erhöhten Entwicklerproduktivität führt.

### Clickable Objects
Die Rezepte sind von nun an anklickbar. Dadurch soll man später nähere Informationen zum Rezept erhalten. Im Moment bekommt man nach anklicken der Rezepte nur die ID angezeigt. Dies könnt ihr euch unten auch nochmal selbst anschauen.
![Clickable](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/b1619886-1d4d-463a-98a6-bbaaa3d48d60)

Demo Video:
https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/54c2386f-52a7-4d8d-86a5-8753c86becd6
Ich hoffe euch hat unser Blogeintrag, wie immer nicht gefallen.
Bis nächste Woche✌️

**Reply by Bjoern04** [reply]
Es ist gut, dass ihr euch dazu entschieden habt zwei Pipelines zu verwenden, da sich dadurch die Qulaität einer Software verbessern wird.
Hattet ihr das Gefühl, dass Verbesserungen in der Kommunikation bei eurer Gruppe nötig sind?

Die Anklickbarkeit der Rezepte scheint eine gute Änderung zu sein, da dort nun die Details der einzelnen Rezepte angezeigt werden können.

Es sieht soweit alles gut aus!

**Reply by amjadalnakshbandi** [reply]
Hallo zusammen,
es sieht so weit gut aus. Wir finden es gut, dass ihr jetzt auch so wie wir Pipelines verwendet und besonders auch, dass ihr uns auch euren Stand mit einem kurzen Video gezeigt habt.

In dem Blogeintrag vom dritten Semester "User Story wurde implementiert: Der Benutzer soll zwischen verschiedenen Rezepten auswählen können", da habt ihr uns auch ein Video von eurem Stand mitgeteilt, der leider von dem Video von gestern kein großer Unterschied hat.
Was wir sehen, es ist jetzt nur die Menüs anklickbar und man bekommt die id zurückgeschickt, ist das richtig so?
Wie sieht es aus mit den weiteren User Stories, habt ihr einen Plan dafür, wann ihr mit den anderen Stories fertig seien werdet?

Viel Erfolg und bis nächste Woche 👍

**Comment by HabubMarcel** [comment]
### Blogeintrag Woche 6 Semester 4: Einführung von neuen Metriken für unser Softwareprojekt

In dieser Woche haben wir uns mit der Auswahl und Implementierung neuer Metriken für unser Projekt beschäftigt. Neben den klassischen Softwaretestmetriken haben wir beschlossen, drei zusätzliche Metriken zu integrieren, um die Qualität und den Fortschritt unseres Projekts besser zu überwachen. Hier sind die Metriken, die wir ausgewählt haben:

#### 1. Mean Time Between Failures (MTBF)
MTBF ist eine wichtige Zuverlässigkeitsmetrik, die die durchschnittliche Zeit zwischen dem Auftreten von Fehlern in unserer Software misst. Sie gibt uns wertvolle Einblicke in die Stabilität und Zuverlässigkeit unseres Systems. Die Berechnung erfolgt wie folgt:
![MTBF](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/0bef26a7-3125-4292-9588-088ed6551bb1)
Durch die Überwachung der MTBF können wir erkennen, wie oft Fehler auftreten und welche Maßnahmen erforderlich sind, um die Systemstabilität zu verbessern.

#### 2. Effektivität der Tests
Diese Metrik misst, wie gut unsere Tests Fehler in der Software aufdecken. Sie berücksichtigt die Anzahl der gefundenen Fehler und wie kritisch diese sind. Eine hohe Effektivität der Tests zeigt, dass unsere Teststrategie erfolgreich ist und die wichtigsten Probleme frühzeitig identifiziert werden. Wir berechnen die Effektivität der Tests anhand folgender Formel:
![Effektivität der Tests](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/fdeac0be-86e2-47a9-9371-96b6d88a934e)
Indem wir die Effektivität der Tests messen, können wir sicherstellen, dass unser Testprozess robust und umfassend ist.

#### 3. Anzahl der offenen Fehler
Die Anzahl der offenen Fehler ist eine einfache, aber wichtige Metrik, die angibt, wie viele bekannte Probleme oder Fehler im Projekt noch nicht behoben wurden. Die Berechnung ist sehr direkt: Sie zählen einfach die Anzahl der offenen Fehler in Ihrem Fehlerverfolgungssystem oder Ihrer Fehlerdatenbank. Im Moment beträgt die Anzahl der Fehler Null.

### Zusammenfassung der studierten Metriken

Durch die Implementierung und regelmäßige Überwachung dieser Metriken können wir die Qualität unseres Projekts kontinuierlich verbessern und sicherstellen, dass wir ein stabiles, fehlerfreies und benutzerfreundliches Produkt liefern.

Ich hoffe, euch hat unser Blogeintrag diese Woche gefallen. Bis nächste Woche✌️

**Reply by amjadalnakshbandi** [reply]
Euer Blogeintrag zu den neuen Metriken ist sehr informativ. Die Einführung von MTBF, Testeffektivität und der Anzahl offener Fehler bietet wertvolle Einblicke zur Verbesserung der Softwarequalität. Gut gemacht!

**Reply by Bjoern04** [reply]
Euer Blogeintrag ist sehr ausführlich und erklärt die drei Metriken und wie man sie misst gut.
Vor allem die Mean Time Between Failure Metrik scheint sehr geeignet zu sein, um die Zuverlässigkeit eines Systems messbar und ohne Merdeutigkeit anzugeben.
Weiter so!

**Comment by 999Marlon** [comment]
# Blogeintrag Woche 8 Semester 4: Review Protokoll
Tobias, Marcel und Luca haben ein Review-Meeting durchgeführt. Dabei wurde sich auf die MainActivity.kt, die Hauptseite, konzentriert. 

**Datum:** 26.05.2024 14 Uhr

**Teilnehmer:** Luca Knaus, Tobias Maaßen (Dokumentieren), Marcel Rohde (Zeitwächter)

**Ziel/Schwerpunkt:** Evaluation der Skalierbarkeit und Wartbarkeit

**Komponenten für den Review:** Hauptseite

**Kriterien für den Review:** Codequalität, Skalierbarkeit und Wartbarkeit 

**Review-Methodik:** Walkthrough 

**Ergebnisse:**
-	Es ist Skalierbar, jedoch abhängig von der Performance des Servers
-	Das anzeigen der Rezepte ist abhängig von der Performance der Datenbank
-	Codequalität war unbefriedigend, diese wurde anschließend verbessert

**Reply by cool2722** [reply]
Schön dass eure Codequalität nun noch verbessert ist, vielleicht könnt ihr aber genauer erläutern was ihr Reviewed haben.

**Reply by Bjoern04** [reply]
Ein gutes Review Meeting. 
Möglicherweise hättet ihr noch einen externen Zuschauer am Meeting teilhaben lassen können. Dieser könnte noch neue Ideen/ Anmerkungen dazu haben.
Habt ihr schon alle bei der Review bemerkten Probleme behoben? Wenn nicht könntet ihr noch Aufgaben zum Beheben der Probleme definieren und den einzelnen Teammitgliedern zuweisen.
Sonst sieht alles gut aus, weiter so!

**Comment by HabubMarcel** [comment]
# Blogeintrag Woche 7 Semester 4
##  CI/CD Pipeline
Willkommen zu unserem Blogeintrag.
Diese Woche haben wir unsere CI/CD Pipelines eingerichtet.
![Pipelines](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/a75bdb1d-5532-44c8-91fe-0aaa71755fa5)
![Badges](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/d8ac4bb9-c1ca-4183-aefd-9e451d38b8ac)
Wir haben eine Unittest Pipeline und eine Android Pipeline. Die Vorgänge in der Pipeline könnt ihr unten in den Bildern einsehen.
![Unittest Pipeline](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/763568dd-2de2-4663-953b-7c5797fc30c0)
![Android Pipeline](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/6e835a29-6a20-47e7-93f0-3e1688eaceb5)
Den YML Code für beide Pipelines findet ihr in der nächsten Zeile.
![Unittest YML](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/88eee406-2187-4220-89f8-db985251ff66)
![Android YML](https://github.com/Qiriba/Rezepte-for-Dummies/assets/149069670/9f100a54-ddbe-4bc7-963a-3a40b81c498e)
Wie ihr sehen könnt, laufen alle Unittests problemlos durch, und unsere Android APP wird gebaut.
### Vorteile
* Automatische Test
* Bit Verifikation
* Spart Zeit

Ich hoffe, euch hat unser Blogeintrag diese Woche gefallen. Bis nächste Woche✌️

**Reply by amjadalnakshbandi** [reply]
Hallo zusammen, 
ihr habt alle erforderlichen Schritte zum Einrichten der CI/CD-Pipeline abgeschlossen. Die GitHub Actions-Konfigurationsdateien für Unit-Tests und Android CI sind gut definiert und die Pipeline umfasst alle erforderlichen Schritte vom Erstellen der Anwendung bis zum Ausführen von Tests. 

Gute Arbeit, Jungs! 👏

**Comment by Qiriba** [comment]
# Blogeintrag Woche 7 Semester 4

Diese Woche haben wir eine Projekt-Retrospektive durchgeführt. Die Farben für die sticky Notes sind: Gelb: Tobias Türkis: Luca Orange: Marcel

### Unsere Dinge die wir Weiter so machen möchten:

![grafik](https://github.com/Qiriba/Rezepte-for-Dummies/assets/136338149/9e88d0ed-5a80-47d9-9dcd-e71ead511eec)


### Davon möchten wir mehr machen 

![grafik](https://github.com/Qiriba/Rezepte-for-Dummies/assets/136338149/a71be8da-84b0-4076-93fa-5050789f2805)


### Hiermit wollen wir Anfangen

![grafik](https://github.com/Qiriba/Rezepte-for-Dummies/assets/136338149/050de5ac-1a05-49c1-8aea-59b786afb790)

### Hiermit Aufhören

![grafik](https://github.com/Qiriba/Rezepte-for-Dummies/assets/136338149/7c04e756-6bd8-47bb-9ce5-f3d4879659cc)

### Und davon weniger

![grafik](https://github.com/Qiriba/Rezepte-for-Dummies/assets/136338149/4e268dfd-bcf6-4269-abb2-37f5ab1ad7ec)

Alles in allem sind wir sehr zufrieden mit dem Ergebniss

**Reply by amjadalnakshbandi** [reply]
Klasse Arbeit geleistet 👏

**Reply by Bjoern04** [reply]
Das sieht sehr gut aus. Jedes Gruppenmitglied hat etwas beigetragen.
Es ist gut, dass ihr euch so reflektierend konnten und eure Probleme erkannt habt und versucht diese in nächster Zeit zu beheben, indem ihr euch z.B. mehr Feedback gebt und Jira mehr einbindet.
Weiter so!
